src/optim/base.py

```diff
@@ -50,6 +52,12 @@ def train_base(model, opt, data, data_seed, scheduler, iterations, acc_steps, ba
     t0 = time.time()
     train_epochs = 0
     while itr < iterations:
+        if not rng_state_dict is None:
```

Comment:
this should only happen in the first execution of the while and not in the next ones. can you move it outside the while?